wayland-client/src/event_queue.rs

```diff
@@ -730,3 +730,32 @@ macro_rules! delegate_dispatch {
         )*
     };
 }
+
+/// A helper macro which delegates a set of [`Dispatch`] implementations for proxies to a no-op handler.
+///
+/// # Usage
+///
+/// This macro is useful to implement [`Dispatch`] for interfaces where events are unimportant to
+/// the current application and can be ignored.
+///
+/// ```
+/// use wayland_client::{delegate_noop, protocol::wl_subcompositor};
+/// delegate_noop!(ExampleApp: [wl_subcompositor::WlSubcompositor]);
+/// ```
+#[macro_export]
+macro_rules! delegate_noop {
+    ($dispatch_from: ty: [ $($interface: ty),* $(,)?]) => {
+        $(
+            impl $crate::Dispatch<$interface, ()> for $dispatch_from {
+                fn event(
+                    _: &mut Self,
+                    _: &$interface,
+                    _: <$interface as $crate::Proxy>::Event,
+                    _: &(),
+                    _: &$crate::Connection,
+                    _: &$crate::QueueHandle<Self>,
+                ) {}
```

Comment:
The implementation should actually have an `unreachable!()`.

One version of the protocol may have no events but a newer version could.

Reply:
I also use this macro for interfaces which do have events, but I don't care. For example, an application that can copy, but not paste won't care about data_offer events.

Perhaps two different macros would make sense (one no-op, one unreachable)?

Reply:
Optionally, the macro could take an `expr` to run, where one could specify `{}` or `unreachable!()` or any other behaviour (e.g.: `println(...`).